src/platform/pico/uart_channel.c

```diff
@@ -0,0 +1,240 @@
+#define UART_CHANNEL_WARN(fmt, ...) LF_WARN(NET, "UartPolledChannel: " fmt, ##__VA_ARGS__)
+#define UART_CHANNEL_INFO(fmt, ...) LF_INFO(NET, "UartPolledChannel: " fmt, ##__VA_ARGS__)
+#define UART_CHANNEL_DEBUG(fmt, ...) LF_DEBUG(NET, "UartPolledChannel: " fmt, ##__VA_ARGS__)
+
+extern Environment *_lf_environment;
+
+UartPolledChannel *uart_channel_0 = NULL;
+UartPolledChannel *uart_channel_1 = NULL;
+FederateMessage msg;
+
+static lf_ret_t UartPolledChannel_open_connection(NetworkChannel *untyped_self) {
+  UartPolledChannel *self = (UartPolledChannel *)untyped_self;
+  char connect_message[] = {0x7, 0x7, 0x7, 0x7};
+  uart_write_blocking(self->uart_device, (const uint8_t *)connect_message, 4);
+  UART_CHANNEL_DEBUG("Open connection");
+  return LF_OK;
+}
+
+static void UartPolledChannel_close_connection(NetworkChannel *untyped_self) {
+  UART_CHANNEL_DEBUG("Close connection");
+  (void)untyped_self;
+}
+
+static void UartPolledChannel_free(NetworkChannel *untyped_self) {
+  UART_CHANNEL_DEBUG("Free");
+  (void)untyped_self;
+}
+
+static bool UartPolledChannel_is_connected(NetworkChannel *untyped_self) {
+  UartPolledChannel *self = (UartPolledChannel *)untyped_self;
+  UART_CHANNEL_DEBUG("Open connection - Sending Ping");
+
+  interval_t wake_up_time = _lf_environment->get_physical_time(_lf_environment) + MSEC(500);
+  lf_ret_t wait_return = LF_SLEEP_INTERRUPTED;
+  while (wait_return == LF_SLEEP_INTERRUPTED) {
+    wait_return = _lf_environment->wait_until(_lf_environment, wake_up_time);
+  }
+  return self->state == NETWORK_CHANNEL_STATE_CONNECTED;
+}
+
+static lf_ret_t UartPolledChannel_send_blocking(NetworkChannel *untyped_self, const FederateMessage *message) {
+  UartPolledChannel *self = (UartPolledChannel *)untyped_self;
+  int message_size = serialize_to_protobuf(message, self->send_buffer, UART_CHANNEL_BUFFERSIZE);
+  UART_CHANNEL_DEBUG("Sending Message of Size: %i", message_size);
+  uart_write_blocking(self->uart_device, (const uint8_t *)self->send_buffer, message_size);
+  return LF_OK;
+}
+
+static void UartPolledChannel_register_receive_callback(NetworkChannel *untyped_self,
+                                                        void (*receive_callback)(FederatedConnectionBundle *conn,
+                                                                                 const FederateMessage *message),
+                                                        FederatedConnectionBundle *bundle) {
+  UART_CHANNEL_DEBUG("Register receive callback");
+  UartPolledChannel *self = (UartPolledChannel *)untyped_self;
+
+  self->receive_callback = receive_callback;
+  self->bundle = bundle;
+}
+
+void _UartPolledChannel_interrupt_handler(UartPolledChannel *self) {
+  while (uart_is_readable(self->uart_device)) {
+    uint8_t received_byte = uart_getc(self->uart_device);
+    self->receive_buffer[self->receive_buffer_index] = received_byte;
+    self->receive_buffer_index++;
```

Comment:
We should check for overflow, that we don't increment receive_buffer_index beyond the max size.